judo-runtime-core-jsl-itest/models/MappedTransferPrimitivesModel/src/test/java/hu/blackbelt/judo/runtime/core/jsl/transfer/MappedTransferQueryCustomizerFunctionsTest.java

```diff
@@ -324,6 +325,274 @@ public void testOrderByDescending() {
+        assertOrderBy(TransferOptionalPrimitivesAttribute.DERIVED_DATE_ATTR, transf2);
+        assertOrderBy(TransferOptionalPrimitivesAttribute.DERIVED_TIMESTAMP_ATTR, transf2);
+        assertOrderBy(TransferOptionalPrimitivesAttribute.DERIVED_TIME_ATTR, transf2);
+        assertOrderBy(TransferOptionalPrimitivesAttribute.DERIVED_ENUM_ATTR, transf2);
+    }
+
+    @Test
+    @Requirement(reqs = {
+            "REQ-MDL-001",
+            "REQ-MDL-002",
+            "REQ-MDL-003",
+            "REQ-TYPE-001",
+            "REQ-TYPE-002",
+            "REQ-TYPE-004",
+            "REQ-TYPE-005",
+            "REQ-TYPE-006",
+            "REQ-TYPE-007",
+            "REQ-TYPE-008",
+            "REQ-TYPE-009",
+            "REQ-ENT-001",
+            "REQ-ENT-002",
+            "REQ-ENT-008",
+            "REQ-SRV-002",
+    })
+    public void testDerivedOrderByDescending() {
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_INTEGER_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_SCALED_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_STRING_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_REGEX_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_BOOL_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_DATE_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_TIMESTAMP_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_TIME_ATTR, transf1);
+        assertOrderByDescending(TransferOptionalPrimitivesAttribute.DERIVED_ENUM_ATTR, transf1);
+    }
+
+    @Test
+    @Requirement(reqs = {
+            "REQ-MDL-001",
+            "REQ-MDL-002",
+            "REQ-MDL-003",
+            "REQ-TYPE-001",
+            "REQ-TYPE-002",
+            "REQ-TYPE-004",
+            "REQ-TYPE-005",
+            "REQ-TYPE-006",
+            "REQ-TYPE-007",
+            "REQ-TYPE-008",
+            "REQ-TYPE-009",
+            "REQ-ENT-001",
+            "REQ-ENT-002",
+            "REQ-SRV-002",
+    })
+    public void testMultiPrimitiveOrderBy() {
+        TransferOptionalPrimitives orderBy = transferOptionalPrimitivesDao
+                .query()
+                .orderBy(TransferOptionalPrimitivesAttribute.INTEGER_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.SCALED_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.REGEX_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.BOOL_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.DATE_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.TIMESTAMP_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.TIME_ATTR)
+                .orderBy(TransferOptionalPrimitivesAttribute.ENUM_ATTR)
```

Comment:
I don't think stacking orderBy-s is the proper solution. Why didn't you tested these orders separately?
If I understood the test code correctly, the order of for example date attributes affects or are affected by other attribute orders leading to possible false successful test results (if for some reason date ordering brakes).

Reply:
This is a specific use for multiply ordering. In esm-itest have test for multiply ordernig. We have already a test case for test order separatley.